File: 3351-maximize-happiness-of-selected-children/maximize-happiness-of-selected-children.cpp
class Solution {
public:
    long long maximumHappinessSum(vector<int>& happiness, int k) {
        sort(happiness.rbegin(),happiness.rend());
        int count =0;
        long long ans =0;
        int n = happiness.size();
        for(int i =0;i<n;i++){
            if(happiness[i]> count && k>0){
                ans += (happiness[i]-count);
                k--;
                count++;
            }
        }
        return ans;
        
    }
};

File: 3351-maximize-happiness-of-selected-children/maximize-happiness-of-selected-children.java
class Solution {
    public long maximumHappinessSum(int[] h, int k) {
        Arrays.sort(h);
        int count =0;
        long ans =0;
        int n = h.length;
        for(int i=n-1;i>=0;i--){
            if(h[i]>count && k >0){
                ans += (h[i]-count);
                count++;
                k--;
            }
        }
        return ans;

        
    }
}